spec/models/guest_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+RSpec.describe Guest, type: :model do
+  describe 'relationships' do
+    it {should have_many :guest_rooms}
+    it {should have_many(:rooms).through(:guest_rooms)}
+  end
+
+  describe 'class methods' do
+    it "can count the total number of guests" do
+      matt = Guest.create!(name: 'Matt', nights: 5)
+      keith = Guest.create!(name: 'Keith', nights: 2)
+      stephanie = Guest.create!(name: 'Stephanie', nights: 4)
+
+      expect(Guest.count_guests).to eq(3)
+
+      sunita = Guest.create!(name: 'Sunita', nights: 4)
+
+      expect(Guest.count_guests).to eq(4)
+    end
+
+    it "return a list of unique guests" do
+      matt = Guest.create!(name: 'Matt', nights: 5)
+      keith = Guest.create!(name: 'Keith', nights: 2)
+      stephanie = Guest.create!(name: 'Stephanie', nights: 4)
+
+      expect(Guest.uniq_guests).to eq([matt, keith, stephanie])
+
+      sunita = Guest.create!(name: 'Sunita', nights: 4)
+
+      expect(Guest.uniq_guests).to eq([matt, keith, stephanie, sunita])
```

Comment:
Similarly to above, the extension asks for this list to be all the guests that stayed _at this Hotel_. I think having an instance of Hotel that we've added some Rooms to and then some Guests to those Rooms, would make this model test complete for the way you designed its functionality. 

Alternatively, you could have made this an instance method on Hotel - and since you gave it another association of `has_many :guests, through: :rooms`, it makes the method very simple to write: 
```
guests.distinct
```
And if you make it this instance method, it makes the testing a lot easier as well. 

So, be careful that your model tests reflect the way you have actually _implemented_ the methods, and not just the way it _can_ work.

Reply:
Also, from your question in the reflection, this test doesn't quite test for "uniqueness"... Consider what the user story is asking for. A hotel should be able to return a unique list of its guests. That means, we'd need to set up a scenario where a guest has stayed at this hotel _more than once_; and in that scenario,t he list would only return that name once. 
This is another argument for having this method be an instance method of Hotel (see my comment above).